plugins/outputs/opensearch/opensearch.go

```diff
@@ -0,0 +1,458 @@
+						"type": "keyword"
+					}
+				}
+			},
+			{
+				"metrics_long": {
+					"match_mapping_type": "long",
+					"mapping": {
+						"type": "float",
+						"index": false
+					}
+				}
+			},
+			{
+				"metrics_double": {
+					"match_mapping_type": "double",
+					"mapping": {
+						"type": "float",
+						"index": false
+					}
+				}
+			},
+			{
+				"text_fields": {
+					"match": "*",
+					"mapping": {
+						"norms": false
+					}
+				}
+			}
+		]
+	}
+}`
+
+type templatePart struct {
+	TemplatePattern string
+	Version         int
+}
+
+func (*Opensearch) SampleConfig() string {
+	return sampleConfig
+}
+
+func (a *Opensearch) Connect() error {
+	if a.URLs == nil || a.IndexName == "" {
+		return fmt.Errorf("opensearch urls or index_name is not defined")
+	}
+
+	// Determine if we should process NaN and inf values
+	switch a.FloatHandling {
+	case "", "none":
+		a.FloatHandling = "none"
+	case "drop", "replace":
+	default:
+		return fmt.Errorf("invalid float_handling type %q", a.FloatHandling)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(a.Timeout))
+	defer cancel()
+
+	var clientOptions []elastic.ClientOptionFunc
+
+	tlsCfg, err := a.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+	tr := &http.Transport{
+		TLSClientConfig: tlsCfg,
+	}
+
+	httpclient := &http.Client{
+		Transport: tr,
+		Timeout:   time.Duration(a.Timeout),
+	}
+
+	osURL, err := url.Parse(a.URLs[0])
+	if err != nil {
+		return fmt.Errorf("parsing URL failed: %v", err)
+	}
+
+	clientOptions = append(clientOptions,
+		elastic.SetHttpClient(httpclient),
+		elastic.SetSniff(a.EnableSniffer),
+		elastic.SetScheme(osURL.Scheme),
+		elastic.SetURL(a.URLs...),
+		elastic.SetHealthcheckInterval(time.Duration(a.HealthCheckInterval)),
+		elastic.SetHealthcheckTimeout(time.Duration(a.HealthCheckTimeout)),
+		elastic.SetGzip(a.EnableGzip),
+	)
+
+	if a.Username != "" && a.Password != "" {
+		clientOptions = append(clientOptions,
+			elastic.SetBasicAuth(a.Username, a.Password),
+		)
+	}
+
+	if a.AuthBearerToken != "" {
+		clientOptions = append(clientOptions,
+			elastic.SetHeaders(http.Header{
+				"Authorization": []string{fmt.Sprintf("Bearer %s", a.AuthBearerToken)},
+			}),
+		)
+	}
+
+	if time.Duration(a.HealthCheckInterval) == 0 {
+		clientOptions = append(clientOptions,
+			elastic.SetHealthcheck(false),
+		)
+		a.Log.Debugf("Disabling health check")
+	}
+
+	client, err := elastic.NewClient(clientOptions...)
+
+	if err != nil {
+		return err
+	}
+
+	// check for OS version on first node
+	osVersion, err := client.ElasticsearchVersion(a.URLs[0])
+
+	if err != nil {
+		return fmt.Errorf("opensearch version check failed: %s", err)
+	}
+
+	// quit if ES version is not supported
+	majorReleaseNumber, err := strconv.Atoi(strings.Split(osVersion, ".")[0])
+	if err != nil || majorReleaseNumber < 1 {
+		return fmt.Errorf("opensearch version not supported: %s", osVersion)
+	}
+
+	a.Log.Infof("Opensearch version: %q", osVersion)
```

Comment:
This check seems useless as there are no 0.x.x versions of OpenSearch.

Reply:
The template isn't doing anything with it? Or am I wrong?

Reply:
yes actually, removed version.

Reply:
As shown in my new review, the version is still requested from OpenSearch.

Reply:
removed version request, log as well.